app/src/main/java/com/humberto/tasky/core/database/data_source/RoomLocalAgendaDataSource.kt

```diff
@@ -128,4 +128,10 @@ class RoomLocalAgendaDataSource(
             else -> Result.Success(Unit)
         }
     }
+
+    override suspend fun deleteAllAgenda() {
+        taskDao.deleteAllTasks()
+        eventDao.deleteAllEvents()
+        reminderDao.deleteAllReminders()
+    }
```

Comment:
There is no need to wait for tasks to be deleted before deleting your events, etc. You can perform each operation in parallel using the following:

```
database.withTransaction {
  withContext(dispatcherProvider.io) {
    listOf(
      async { eventDao.deleteAllEvents() },
      async { reminderDao.deleteAllReminders() },
      async { taskDao.deleteAllTasks() }
    ).awaitAll()
  }
}
```

We call `awaitAll()` to only return this function when all the operations have been completed.
We also wrap the call with `database.WithTransaction` to rollback changes when one of the delete operations fail